src/modules/networkmanager.rs

```diff
@@ -0,0 +1,87 @@
+use color_eyre::Result;
+use futures_lite::StreamExt;
+use futures_signals::signal::SignalExt;
+use gtk::prelude::ContainerExt;
+use gtk::{Box as GtkBox, Image, Orientation};
+use serde::Deserialize;
+use tokio::sync::mpsc::Receiver;
+
+use crate::clients::networkmanager::{Client, ClientState};
+use crate::config::CommonConfig;
+use crate::gtk_helpers::IronbarGtkExt;
+use crate::image::ImageProvider;
+use crate::modules::{Module, ModuleInfo, ModuleParts, ModuleUpdateEvent, WidgetContext};
+use crate::{glib_recv, module_impl, send_async, spawn};
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct NetworkManagerModule {
```

Comment:
The introduction of JSON schema support annoyingly means every type used in the config needs to derive the `JsonSchema` type when the feature is enabled:

```suggestion
#[derive(Debug, Deserialize, Clone)]
#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
pub struct NetworkManagerModule {
```

Reply:
Added it :+1: